content/en/blog/2022/rust-instrumentation.md

```diff
@@ -0,0 +1,378 @@
+
+For example, if we want to filter some of our data, export it, format it or edit it, we can create or use an existing layer and compose it with the registry subscriber.
+
+```shell
+let subscriber = tracing_subscriber::Registry::default().with(SomeLayer)
+```
+
+We will see more examples on how to use these crates to instrument a Rust program in the practical section.
+
+### OpenTelemetry in Rust
+
+The crate that provides support for OpenTelemetry is called - how convenient - “[OpenTelemetry](https://docs.rs/opentelemetry/latest/opentelemetry/)”.
+
+“OpenTelemetry provides a single set of APIs, libraries, agents, and collector services to capture distributed traces and metrics from your application.”
+
+Let’s go over some of its key APIs:
+
+- Tracer. Inside the crate, there is a module called traces. It introduces us to a trait called Tracer. This is what tracks and connects our spans to create traces. To use Tracer, we create a subscriber layer. This is how the subscriber knows where to send the spans and how to generate traces.
+
+- Global. This module provides an API so that no matter in what section of the code we currently are, we can access the subscriber, tracer, and propagation context.
+
+- SDK. This module provides an implementation for common use cases with OpenTelemetry. For example, it provides an exporter to send traces to stdout. It also implements context propagation methods that we can use instead of implementing ourselves.
+
+The git repository called [opentelemetry-rust](https://github.com/open-telemetry/opentelemetry-rust) contains implementations of several crates that expand the opentelemetry ecosystem. There you can find common instrumentations, exporters, and subscribers use them in a Rust program.
+
+### Distributed tracing in Rust
+
+Using crates from the [opentelemetry-rust](https://github.com/open-telemetry/opentelemetry-rust) repo, we can instrument our application across multiple services and send them to a distributed platform.
+
+For example, we can use the [opentelemetry-jaeger](https://crates.io/crates/opentelemetry-jaeger) to send our traces to a remote Jaeger platform.
+
+Another example is the [OpenTelementy-otlp](https://crates.io/crates/opentelemetry-otlp) crate.
+
+Traces in OTLP format can be exported to the OpenTelemetry Collector using the exporter in this crate. OpenTelemetry Collector accepts, processes, and exports traces in a vendor-agnostic manner.
+
+## Practical Example
+
+For this example, I wanted to build a simple system I could use to integrate and visualize traces. I found auto-instrumentations for [Actix-Web](https://actix.rs/) Framework and [Diesel](https://diesel.rs/), an ORM and Query builder utility for SQL-based databases.
+
+I chose to use these tools and build a simple REST api service to create, read and delete users following the examples given by these frameworks.
+
+Here is a link for the [source code](https://github.com/aspecto-io/opentelemetry-examples).
```

Comment:
point to the specific rust repo, I only see node & go there, also no README what to do exactly with that example.